pygeos/tests/test_strtree.py

```diff
@@ -211,9 +211,45 @@ def test_query_unsupported_predicate(tree):
         tree.query(pygeos.points(1, 1), predicate="disjoint")
 
 
-def test_query_predicate_errors(tree):
+@pytest.mark.parametrize(
+    "predicate,expected",
+    [
+        ("intersects", [0, 1, 2]),
+        ("within", []),
+        ("contains", [1]),
+        ("overlaps", []),
+        ("crosses", []),
+        ("covers", [0, 1, 2]),
+        ("covered_by", []),
+        ("contains_properly", [1]),
+    ],
+)
+def test_query_prepared_inputs(tree, predicate, expected):
+    geom = box(0, 0, 2, 2)
+    pygeos.prepare(geom)
+    assert_array_equal(tree.query(geom, predicate=predicate), expected)
+
+
+@pytest.mark.parametrize(
```

Comment:
Expanded this test to verify that memory leak occurs for all predicates except `intersects`